src/Symfony/Component/Form/ButtonBuilder.php

```diff
@@ -63,6 +63,18 @@ public function __construct(?string $name, array $options = array())
 
         $this->name = $name;
         $this->options = $options;
+
+        if (\preg_match('/^([^a-z0-9_].*)?(.*[^a-zA-Z0-9_\-:].*)?$/D', $name, $matches)) {
```

Comment:
Just asking: is this validating the value of the `name` HTML attribute of `<button>` elements ... or is this a Symfony thing? I say this because HTML allows for example `:` as the first character of `name`.

Reply:
It affects both. The internal name is used to generate the HTML attribute value. Buttons will then behave the same as "normal" form types already behave.